src/kvstore/comm.h

```diff
@@ -170,34 +170,36 @@ class CommCPU : public Comm {
     }
   }
 
-  // TODO(haibin) support broadcast row_sparse on GPU
   void BroadcastRowSparse(int key, const NDArray& src,
                           const std::vector<std::pair<NDArray*, NDArray>>& dst,
                           const bool use_copy,
                           const int priority) override {
     using namespace mshadow;
-    auto size = dst.size();
-    for (size_t i = 0; i < size; i++) {
-      auto out = dst[i].first;
-      auto row_id = dst[i].second;
+    for (size_t i = 0; i < dst.size(); ++i) {
```

Comment:
Function call overhead is relatively high if it doesn't get inlined for some reason. You probably wouldn't notice it in one pass, but if this function gets called millions+ of times during training, it can add up. It probably wouldn't be a bottleneck, but a CPU tick here and a CPU tick there eventually add up to real time. The question is why spend ticks when it's very simple not to?